src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -200,17 +197,50 @@ static void CheckForExperimentStatus(ITextView textView, Document document)
             return true;
         }
 
-        public async Task<AsyncCompletionData.CompletionContext> GetCompletionContextAsync(
+        public Task<AsyncCompletionData.CompletionContext> GetCompletionContextAsync(
+            IAsyncCompletionSession session,
+            AsyncCompletionData.CompletionTrigger trigger,
+            SnapshotPoint triggerLocation,
+            SnapshotSpan applicableToSpan,
+            CancellationToken cancellationToken)
+        {
+            // Have to store the trigger location to reuse it to get expanded items and also reuse the snapshot included in 
+            // some projections related scenarios where data and session in further calls are able to provide other snapshots.
+            session.Properties[TriggerLocation] = triggerLocation;
```

Comment:
prefer this to be handled in session directly not in the property bag

Reply:
Fixed